scripts/west_commands/sdk.py

```diff
@@ -0,0 +1,553 @@
+
+    def sha256_sum_url(self, release):
+        assets = release.get("assets", [])
+        minimal_sdk_asset = next(filter(lambda x: x["name"] == "sha256.sum", assets))
+
+        return minimal_sdk_asset["browser_download_url"]
+
+    def download_and_extract(self, base_dir, dir_name, target_release, req_headers):
+        self.inf("Fetching sha256...")
+        sha256_url = self.sha256_sum_url(target_release)
+        resp = requests.get(sha256_url, headers=req_headers, stream=True)
+        if resp.status_code != 200:
+            raise Exception(f"Failed to download {sha256_url}: {resp.status_code}")
+
+        sha256 = self.minimal_sdk_sha256(resp.content.decode("UTF-8"), target_release)
+
+        archive_url = self.minimal_sdk_url(target_release)
+        self.inf(f"Downloading {archive_url}...")
+        resp = requests.get(archive_url, headers=req_headers, stream=True)
+        if resp.status_code != 200:
+            raise Exception(f"Failed to download {archive_url}: {resp.status_code}")
+
+        try:
+            if not os.path.exists(base_dir):
+                os.makedirs(base_dir)
+
+            with tempfile.TemporaryDirectory(dir=base_dir) as tempdir:
+                # download archive file
+                filename = os.path.join(tempdir, re.sub(r"^.*/", "", archive_url))
+                file = open(filename, mode="wb")
+                total_length = int(resp.headers["Content-Length"])
+                count = 0
+
+                for chunk in resp.iter_content(chunk_size=8192):
+                    file.write(chunk)
+                    count = count + len(chunk)
+                    self.inf(f"\r {count}/{total_length}", end="")
+                self.inf()
+                self.inf(f"Downloaded: {file.name}")
+                file.close()
+
+                # check sha256 hash
+                with open(file.name, "rb") as sha256file:
+                    digest = hashlib.sha256(sha256file.read()).hexdigest()
+                    if sha256 != digest:
+                        raise Exception(f"sha256 mismatched: {sha256}:{digest}")
+
+                # extract archive file
+                self.inf(f"Extract: {file.name}")
+                if file.name.endswith(".tar.xz"):
+                    with tarfile.open(file.name, mode="r:xz") as archive:
+                        archive.extractall(path=tempdir)
+                elif file.name.endswith(".tar.gz"):
+                    with tarfile.open(file.name, mode="r:gz") as archive:
+                        archive.extractall(path=tempdir)
+                elif file.name.endswith(".7z"):
+                    with SevenZipFile(file.name, mode="r") as archive:
+                        archive.extractall(path=tempdir)
+                elif file.name.endswith(".zip"):
+                    with ZipFile(file.name, mode="r") as archive:
+                        archive.extractall(path=tempdir)
+                else:
+                    raise Exception("Unknown file extension.")
```

Comment:
can't you use patool? it feels weird to not rely on an existing well established module to deal with this kind of boilerplate